### PR TITLE
Fix/component state

### PR DESCRIPTION
**Pull request by ameliaeiger**
This one is a doozy. The issue with components being stored in state led me down a rabbit hole. URIs and base strings were mislabeled; each was labeled after the other. Refactored for clarity, adherence to style guidelines. In addition:

Solved two issues that were causing us to be unable to access state accurately:

1. the issue of components being stored within a state
2. the getView function inserting the first uri to the DOM directly without storing to state first

I have rearranged the data flow quite a bit, making it so that just information is stored within state and Components are generated and rendered as needed/where needed. However, most of the end results have remained the same except for a few labels. I didn't feel confident in messing much with the response page, so please check that part out with a cautious eye to ensure everything DID end up in it's proper place.

I also fixed all of those annoying command line errors: 

1. Memory leak caused from font loading on unmounted components due to poor use of useEffect is fixed
2. All child components have keys in the proper places, so no more annoying error message every time a photo is uploaded from native lib

Lemme know if you have any questions!

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -1,63 +1,52 @@
...
 import ResponsePage from './Views/ResponsePage'
-import PlantImage from './components/ImageDisplay/PlantImage'
 import HeaderLogo from './components/Header'
 
-export default function App() {
-  const [imageUris, addImageUris] = useState([])
-  const [allImages, setAllImages] = useState([])
+export default App = () => {
+  const [imageBaseStrings, setImageBaseStrings] = useState([])
+  const [imageUris, setImageUris] = useState([])
```
Comment (BlueNealis) [inline App.js:16]: Nice yes this is much clearer thank you

File: Views/Dashboard.js [hunk 2]
```diff
@@ -1,27 +1,35 @@
...
+import * as Haptics from 'expo-haptics'
+
+//COMPONENTS
+import PlantImage from '../components/ImageDisplay/PlantImage'
 import PickImage from "../components/PickImage/PickImage"
 import TakePicture from "../components/TakePicture/TakePicture"
-import * as Haptics from 'expo-haptics';
 
-export default function Dashboard({ navigation, handleChange, allImages, images, dimensions, apiKey }) {
+export default Dashboard = ({ navigation, handleChange, imageUris, imageBaseStrings, dimensions, apiKey }) => {
```
Comment (BlueNealis) [inline Views/Dashboard.js:14]: One thing Im wondering is are we still using the image Base strings anymore?

File: Views/Dashboard.js [hunk 3]
```diff
@@ -65,18 +72,19 @@ export default function Dashboard({ navigation, handleChange, allImages, images,
...
-            })
-            navigation.navigate("Results", {apiKey: apiKey,
-            uris:images,
-            })
-          }}> discover
+              Haptics.notificationAsync(Haptics.NotificationFeedbackType.Success)
+              .catch(error => {
+                return
+              })
+              //Navigates to ResponsePage, NOT the component Results. However, this is how the title of the results page is displayed to the user.
```
Comment (BlueNealis) [inline Views/Dashboard.js:79]: Sounds good thank you for the comment here

**Review by BlueNealis — APPROVED** [review]
Thank you this definitely makes its more readable as Im looking through this.

File: Views/Dashboard.js (hunk 2 above)
Comment (ameliaeiger) [reply Views/Dashboard.js:14]: okay so what I figured out was: the image URIs are being used to display images on the phone, base strings are being sent off to get info.

**Review by laytonmaes — APPROVED** [review]

**Review by Neil-B-Linton — APPROVED** [review]
Great changes, looks amazing

**Pull request merged**